### PR TITLE
Disallow shouty constants

### LINKED ISSUE #761: New lint rule: Disallow shouty constants that have the same variable name as the value
This is such a bad pattern:

```
const FOO = 'FOO';
```

It's typically used for constants. The prevailing consensus once you actually drill down into why people do this is for no good reason.

We can have an autofix to inline these (unless exported, which we should probably error on).

This should be _relatively_ easy.

 - For all `JSReferenceIdentifier`, look up the `scope.getBinding`. If it matches `binding instanceof ConstBinding && !binding.isExported` and you check the `binding.value` and it's a string literal, with the same value as the name, inline it.
  - Do the reverse and remove `VariableDeclarator` that matches the above heuristics (not exported etc).

(There may be some usages in Rome so I'm a hypocrite.)

### PR DESCRIPTION
**Pull request by ooflorent**
Fixes #761

### CODE REVIEW

File: packages/@romefrontend/diagnostics/descriptions/lint.ts [hunk 1]
```diff
@@ -16,6 +16,10 @@ import {buildSuggestionAdvice} from "../helpers";
 import {createDiagnosticsCategory, orJoin} from "./index";
 
 export const lint = createDiagnosticsCategory({
+	JS_SHOUTY_CONSTANTS: {
+		category: "lint/js/shoutyConstants",
+		message: "TODO",
```
Comment (ooflorent) [inline packages/@romefrontend/diagnostics/descriptions/lint.ts:21]: Could someone suggest a description message?

File: packages/@romefrontend/diagnostics/descriptions/lint.ts [hunk 2]
```diff
@@ -16,6 +16,10 @@ import {buildSuggestionAdvice} from "../helpers";
...
+	},
```
Comment (sebmck) [inline packages/@romefrontend/diagnostics/descriptions/lint.ts:22]:
```suggestion
	JS_SHOUTY_CONSTANTS: (constantLocation: DiagnosticLocation = {}) => ({
		category: "lint/js/shoutyConstants",
		message: "Redundant constant reference",
		advice: [
			{
				type: "log",
				category: "info",
				text: "You should avoid declaring constants with a string that's the same value as the variable name. It introduces a level on unnecessary indirection when it's only two additional characters to inline.",
			},
			{
				type: "log",
				category: "info",
				text: "This constant is declared here",
			},
			{
				type: "frame",
				location: constantLocation,
			},
		],
	}),
```

File: packages/@romefrontend/compiler/lint/rules/js/shoutyConstants.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+				binding.value.value === node.name &&
+				!binding.isExported &&
+				(binding.scope.kind === "block" || binding.scope.kind === "program")
+			) {
+				return path.context.addFixableDiagnostic(
+					{
+						old: node,
+						fixed: binding.value,
+					},
+					descriptions.LINT.JS_SHOUTY_CONSTANTS,
```
Comment (sebmck) [inline packages/@romefrontend/compiler/lint/rules/js/shoutyConstants.ts:24]:
```suggestion
					descriptions.LINT.JS_SHOUTY_CONSTANTS(binding.node.loc),
```

File: packages/@romefrontend/diagnostics/descriptions/lint.ts [hunk 4]
```diff
@@ -17,6 +17,26 @@ import {buildSuggestionAdvice} from "../helpers";
...
 
 export const lint = createDiagnosticsCategory({
+	JS_SHOUTY_CONSTANTS: (constantLocation: DiagnosticLocation = {}) => ({
+		category: "lint/js/shoutyConstants",
+		message: "Redundant constant reference",
+		advice: [
+			{
+				type: "log",
+				category: "info",
+				text: "You should avoid declaring constants with a string that's the same value as the variable name. It introduces a level on unnecessary indirection when it's only two additional characters to inline.",
```
Comment (yassere) [inline packages/@romefrontend/diagnostics/descriptions/lint.ts:27]:
```suggestion
				text: "You should avoid declaring constants with a string that's the same value as the variable name. It introduces a level of unnecessary indirection when it's only two additional characters to inline.",
```
Small typo.

**Comment by sebmck** [comment]
Thank you!

**Pull request merged**